services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/controller/GSController.java

```diff
@@ -0,0 +1,173 @@
+    gs.setRsType(Common.ResourceType.PORT);
+    return service(gs);
+  }
+
+  /**
+   * Accept north bound calls then transfer to ACA calls in southbound
+   *
+   * @param gs Encapsulation of NetworkConfiguration message
+   * @return RestOperationResult in String
+   * @throws Exception Various exceptions that may occur during the update process
+   * @link https://github.com/haboy52581/alcor/blob/master/docs/modules/ROOT
+   *     /pages/infra_services/data_plane_manager.adoc
+   */
+  @PutMapping({"/port/", "v4/port/"})
+  public List<InternalDPMResult> updatePort(@RequestBody NetworkConfiguration gs)
+      throws Exception {
+    gs.setRsType(Common.ResourceType.PORT);
+    gs.setOpType(Common.OperationType.UPDATE);
+    return service(gs);
+  }
+
+  /**
+   * Accept north bound calls then transfer to ACA calls in southbound
+   *
+   * @param gs Encapsulation of NetworkConfiguration message
+   * @return RestOperationResult in String
+   * @throws Exception Various exceptions that may occur during the delete process
+   * @link https://github.com/haboy52581/alcor/blob/master/docs/modules/ROOT
+   *     /pages/infra_services/data_plane_manager.adoc
+   */
+  @DeleteMapping({"/port/", "v4/port/"})
+  public List<InternalDPMResult> deletePort(@RequestBody NetworkConfiguration gs)
+      throws Exception {
+    gs.setOpType(Common.OperationType.DELETE);
+    gs.setRsType(Common.ResourceType.PORT);
+    return service(gs);
+  }
+
+  /**
+   * Accept north bound calls then transfer to ACA calls in southbound
+   *
+   * @param gs Encapsulation of NetworkConfiguration message
+   * @return RestOperationResult in String
+   * @throws Exception Various exceptions that may occur during the create process
+   * @link https://github.com/haboy52581/alcor/blob/master/docs/modules/ROOT
+   *     /pages/infra_services/data_plane_manager.adoc
+   */
+  @PostMapping({"/subnet/", "v4/subnet/"})
+  @ResponseStatus(HttpStatus.CREATED)
+  public List<InternalDPMResult> createSubnet(@RequestBody NetworkConfiguration gs)
+      throws Exception {
+    gs.setOpType(Common.OperationType.CREATE);
+    gs.setRsType(Common.ResourceType.SUBNET);
+    return service(gs);
+  }
+
+  /**
+   * Accept north bound calls then transfer to ACA calls in southbound
+   *
+   * @param gs Encapsulation of NetworkConfiguration message
+   * @return RestOperationResult in String
+   * @throws Exception Various exceptions that may occur during the update process
+   * @link https://github.com/haboy52581/alcor/blob/master/docs/modules/ROOT
+   *     /pages/infra_services/data_plane_manager.adoc
+   */
+  @PutMapping({"/subnet/", "v4/subnet/"})
+  public List<InternalDPMResult> updateSubnet(@RequestBody NetworkConfiguration gs)
+      throws Exception {
+    gs.setOpType(Common.OperationType.UPDATE);
+    gs.setRsType(Common.ResourceType.SUBNET);
+    return service(gs);
+  }
+
+  /**
+   * Accept north bound calls then transfer to ACA calls in southbound
+   *
```

Comment:
Do we have a scenario to use these subnet APIs?

Reply:
since dpm is not only for port but for all network resources, I suggest leave the api here

Reply:
Okay. We can leave some comments to remind that they are not in use for now.